### DISCUSSION TITLE
How to get on-board led pin information?

### DISCUSSION THREAD

**Discussion by wh1tecat-nya** [microsoft/devicescript#411] [Q&A] [answered]
I found a definition of led pin information in boards.json.
https://github.com/microsoft/devicescript/blob/9f8380259428f0f6290c9f7379bdfe3e06250418/compiler/src/boards.json#L695-L697

But it is not included `@dsboard/{boardname}(devicescript-boards.d.ts)`.
How to use on-board led or led pin assignment?

**Comment by pedropalhari** [comment]
Having the same issue, trying to instantiate it directly with `gpio(25)` also fails. Seems to be in use by something else (like `gpio(0)`) but couldn't figure it out why yet.

**Comment by pelikhan** [comment, marked as answer]
For the onboard LED, you would need to use `setStatusLight` --> https://microsoft.github.io/devicescript/developer/status-light . The onboard LED is also used to notify the status of the network stack and is "shared" with the runtime.

Keep this issue opened to beef up the docs.

**Reply by pelikhan** [reply]
Note that the simulator is also now showing the state of setStatusLight.

**Comment by pedropalhari** [comment]
Hi, @pelikhan, thanks for the follow up!

Weirdly, I did encounter that section of the docs (and tried it). And on the RPi 2040 at least it's not (currently) working. It compiles and runs with no issue but nothing happens on the board. I tested on 2 different RPis I have here.

I was able to simulate a "hello world" with an external led though and didn't explore it much more (was just doing a quick getting started to check the framework) so I can't provide additional info now. But I can come back to it during the weekend and see if I'm able to pinpoint exactly what's happening.

**Reply by pelikhan** [reply]
@pedropalhari are you using the pico-w , the onboard LED is wired differently (through the wifi chip) and we haven't implemented support for it yet.

**Reply by pedropalhari** [reply]
Actually no, I'm using the regular one.

I also noticed that the sample code for `Controlling the status led` at https://microsoft.github.io/devicescript/developer/status-light#controlling-the-status-led does a `setInterval` with the same period as the `delay` inside of it, which causes a weird overlap. 

Anyway, if I do:

```ts
import { setStatusLight } from "@devicescript/runtime"

console.log("[LED] Before led");
await setStatusLight(0xffffff);
console.log("[LED] After led");
```

Both consoles get printed but nothing happens on the board.

**Reply by pedropalhari** [reply]
I know this is too little information to help debugging, I'd help with a more in-depth look at the framework but I'm currently at work

**Reply by pelikhan** [reply]
No worries, we're preparing for a 3day weekend here in the US so it's mostly gonna be for next week :)

**Reply by pelikhan** [reply]
The setinterval example should be fine as we await the function. However, it might be exactly in sync with the jacdac blinking. The `setStatusLight` semantics is that it shares the LED with other system "status" which may actually more confusing than useful.

**Reply by pedropalhari** [reply]
Understood, could be that some event is overwriting my setStatusLight call then. What I have on my end is the led not turning on at all on RPi 2040 (non W).

**Reply by pelikhan** [reply]
Indeed `setStatusLight` on pico is not giving signs of life. @mmoskal did we implement setStatusLight on pico already?

**Reply by pelikhan** [reply]
Tracking with https://github.com/microsoft/devicescript/issues/404

**Comment by pelikhan** [comment]
Improved docs with https://github.com/microsoft/devicescript/commit/ecb7c7be5ab9c882e6194880e63d4ba60932fd06
